transport.go

```diff
@@ -142,19 +147,21 @@ func (log *HcLogToLogger) ImpliedArgs() []interface{} {
 // streamLayer an implementation of raft.StreamLayer for use
 // with raft.NetworkTransportConfig.
 type streamLayer struct {
-	host host.Host
-	l    net.Listener
+	host     host.Host
+	l        net.Listener
+	protocol protocol.ID
 }
 
-func newStreamLayer(h host.Host) (*streamLayer, error) {
-	listener, err := gostream.Listen(h, RaftProtocol)
+func newStreamLayer(h host.Host, protocol protocol.ID) (*streamLayer, error) {
```

Comment:
Let's use a protocol _prefix_. That way we can upgrade the protocol over time while retaining the prefix (this is how we do it in bitswap, the DHT, etc.).